packages/utils/upgrade/src/core/requirements/is-clean-git-repo.ts

```diff
@@ -0,0 +1,39 @@
+import assert from 'node:assert';
+import { simpleGit } from 'simple-git';
+import type { TaskOptions } from '../../types';
+
+type Params = Pick<TaskOptions, 'confirm' | 'logger'> & {
+  force: boolean;
+  cwd: string;
+};
+
+export const isCleanGitRepo = async ({ cwd, logger, force, confirm }: Params) => {
+  const git = simpleGit({ baseDir: cwd });
```

Comment:
According to their docs, git is a pre-req. Could you add an extra check for `hasGitInstalled`?